### PR TITLE
Add changelog generator to workflow

### PR DESCRIPTION
**Pull request by ccremer**
Adds a changelog generator to the project.

Each PR from this point on would need a label in order to show up in the changelog
The release notes can be edited afterwards ofc, if there's additional release notes to add.

No "prepare release" PRs would have to be made just to update the previous CHANGELOG.md

Example release in my temporary fork: https://github.com/ccremer/commodore/releases/tag/v1.0.0

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [ ] ~Update the ./CHANGELOG.md.~ -> Removed
- [ ] Link this PR to related issues.

### CODE REVIEW

**Review by simu — CHANGES_REQUESTED** [review]
LGTM overall, but needs cleanup of references to `CHANGELOG.md` in `README.md` and `docs/modules/ROOT/pages/reference/deprecation-notices.adoc`

**Review by srueg — APPROVED** [review]
Very nice!

Can we also update the component template accordingly? https://github.com/projectsyn/commodore/tree/master/commodore/component-template/%7B%7B%20cookiecutter.slug%20%7D%7D

File: commodore/component-template/{{ cookiecutter.slug }}/.github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -12,7 +12,9 @@ Remove items that do not apply. For completed items, change [ ] to [x].
 
 - [ ] Keep pull requests small so they can be easily reviewed.
 - [ ] Update the documentation.
-- [ ] Update the ./CHANGELOG.md.
+- [ ] Categorize the PR by setting a good title and adding one of the labels:
+      `bug`, `enhancement`, `documentation`, `change`, `breaking`,
```
Comment (srueg) [inline commodore/component-template/{{ cookiecutter.slug }}/.github/PULL_REQUEST_TEMPLATE.md:16]:
```suggestion
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -13,7 +13,9 @@ Remove items that do not apply. For completed items, change [ ] to [x].
 - [ ] Keep pull requests small so they can be easily reviewed.
 - [ ] Update the documentation.
 - [ ] Update tests.
-- [ ] Update the ./CHANGELOG.md.
+- [ ] Categorize the PR by setting a good title and adding one of the labels:
+      `bug`, `enhancement`, `documentation`, `change`, `breaking`,
```
Comment (srueg) [inline .github/PULL_REQUEST_TEMPLATE.md:17]:
```suggestion
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
```

**Review by srueg — APPROVED** [review]

File: commodore/component-template/{{ cookiecutter.slug }}/.github/changelog-configuration.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{
+    "pr_template": "- ${{TITLE}} (#${{NUMBER}})",
```
Comment (srueg) [inline commodore/component-template/{{ cookiecutter.slug }}/.github/changelog-configuration.json:2]: Seems like this trips up the jinja template of cookiecutter... You might have to escape the `{{` and `}}`

Comment (ccremer) [reply commodore/component-template/{{ cookiecutter.slug }}/.github/changelog-configuration.json:2]: can I test it out locally?

Comment (ccremer) [reply commodore/component-template/{{ cookiecutter.slug }}/.github/changelog-configuration.json:3]: I just `{% raw %}`d the whole file :)

**Review by srueg — APPROVED** [review]

**Review by simu — APPROVED** [review]
LGTM, for completeness sake you could update the list of files expected to exist after running `commodore component new` in `tests/test_component_template.py`.

**Comment by ccremer** [comment]
> LGTM, for completeness sake you could update the list of files expected to exist after running `commodore component new` in `tests/test_component_template.py`.

All .github files were missing, so I added those. Did you mean this?

**Comment by simu** [comment]
> > LGTM, for completeness sake you could update the list of files expected to exist after running `commodore component new` in `tests/test_component_template.py`.
> 
> All .github files were missing, so I added those. Did you mean this?

Yeah, looks like that test did not really get updated when we changed the template. Thanks!

**Review by simu — APPROVED** [review]

**Pull request merged**